packages/block-library/src/navigation-link/edit.js

```diff
@@ -193,6 +222,14 @@ function NavigationLinkEdit( {
 					/>
 				</PanelBody>
 			</InspectorControls>
+			<BlockNavigationListItemFill>
+				<BlockNavigationListItem
+					wrapperComponent="div"
+					onClick={ noop }
+				>
+					{ editField }
+				</BlockNavigationListItem>
+			</BlockNavigationListItemFill>
```

Comment:
Given the way the design is going, I'm wondering if we should consider a more straightforward component that only makes it possible to edit text. Internally I think this would still use the fill, but externally it'd just be a single component that makes it possible edit the text for a block:
```jsx
// Not sure on the naming here, but to illustrate:
<BlockNavigationEdit
  onChange={ ( labelValue ) => setAttributes( { label: labelValue } ) }
  value={ value }
/>
```

Explaining my thought process—originally when we looked at using a slot/fill I wasn't sure if there would be additional functionality built in to a single block in the block navigation—for example, being able to edit the href of a navigation link. That doesn't seem to be the case now, any functionality like that (if it is added) seems like it'll end up in the dropdown menu shown in https://github.com/WordPress/gutenberg/issues/22089.

I think there's a benefit in that this gives more control over how the block navigation will look and what features are exposed (for example, more control over handling formatting).

What do you think?

Reply:
@talldan yup it makes sense, I just adjusted the PR